src/main.rs

```diff
@@ -99,7 +100,7 @@ fn main() {
 
         let mut types = vec![DiagnosticClass::Parse];
 
-        if !matches.is_present("discouraged") {
+        if !matches.is_present("discouraged") && !matches.is_present("axiom_use") {
```

Comment:
So if the tool was called with both axiom use and verify options, like `-X axiom_use -v`, the verify would not report scope issues?
I don't understand why we have these tests.

Reply:
If you run a semantically complex pass, it will automatically perform all the prerequisites first anyway. So if you did `-X -v` then the scope pass would run. The only reason this if statement is needed is because almost everything requires the scope pass so it is "enabled by default", unless you do discouraged / axiom_use generation since they work on the text itself and so only require the basic parser pass.

Reply:
Yes the scope pass would run, but the generated errors would not be displayed.

The `types` vector is used by `diag_notations`. In case the `Scope` enum is not set, by [the following lines](https://github.com/david-a-wheeler/metamath-knife/blob/97c28613cb0a07da62bcebc67a9d4af39192712d/src/database.rs#L913), the `diag` list of diagnostics would include scope errors.
```
        if types.contains(&DiagnosticClass::Scope) {
            diags.extend(self.scope_pass().diagnostics());
        }
```

Reply:
Hm, you're right. I don't really like that architecture, the same issue arises for `-p` since the `StmtParse` pass depends on the `Grammar` pass but if you enable only `StmtParse` then the grammar pass will run and errors will not be reported.

Better would be for it to just report all errors from passes that have run, regardless of whether they were run en passant to some other pass.

Reply:
This is apparently by design, according to the doc comment:
```rust
    /// Collects and returns all errors generated by the passes run.
    ///
    /// Passes are identified by the `types` argument and are not inclusive; if
    /// you ask for Verify, you will not get Parse unless you specifically ask
    /// for that as well.
    ///
    /// Currently there is no way to incrementally fetch diagnostics, so this
    /// will be a bit slow if there are thousands of errors.
    pub fn diag_notations() { ... }
```
Should I get rid of this behavior?

Reply:
Ok, I see you already have a different PR for that!